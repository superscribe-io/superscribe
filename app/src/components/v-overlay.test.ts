import { test, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import VOverlay from './v-overlay.vue';

test('Mount component', () => {
	expect(VOverlay).toBeTruthy();

	const wrapper = mount(VOverlay, {
		props: {
			active: true,
		},
		slots: {
			default: 'Slot Content',
		},
	});

	expect(wrapper.html()).toMatchSnapshot();
});

test('absolute prop', () => {
	const wrapper = mount(VOverlay, {
		props: {
			absolute: true,
		},
	});

	expect(wrapper.classes()).toContain('absolute');
});

test('clickable prop', () => {
	const wrapper = mount(VOverlay, {
		props: {
			clickable: true,
		},
	});

	expect(wrapper.classes()).toContain('has-click');
});
