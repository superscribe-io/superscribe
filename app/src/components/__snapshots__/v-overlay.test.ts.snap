// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Mount component 1`] = `
"<div data-v-500f2fb5=\\"\\" class=\\"v-overlay active has-click\\">
  <div data-v-500f2fb5=\\"\\" class=\\"overlay\\"></div>
  <div data-v-500f2fb5=\\"\\" class=\\"content\\">Slot Content</div>
</div>"
`;
